{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 90,
   "source": [
    "import pandas as pd\r\n",
    "from tqdm import tqdm\r\n",
    "tqdm.pandas()"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "source": [
    "df = pd.read_csv('promed_malaria.csv')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "source": [
    "def clean(content):\r\n",
    "\tsplit = content.splitlines()\r\n",
    "\tlast_index = -1\r\n",
    "\tlower = [x.lower().strip() for x in split]\r\n",
    "\tif '--' in lower:\r\n",
    "\t\tlast_index = lower.index('--')\r\n",
    "\telif 'communicated by:' in lower:\r\n",
    "\t\tlast_index = lower.index('communicated by:')-1\r\n",
    "\r\n",
    "\tcleaned = split[12:last_index]\r\n",
    "\treturn '\\n'.join([x for x in cleaned if x])"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "source": [
    "df['content'] = df['content'].apply(clean)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "source": [
    "import sys\r\n",
    "sys.path.append('../EpiTator')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "source": [
    "from epitator.annotator import AnnoDoc\r\n",
    "from epitator.count_annotator import CountAnnotator\r\n",
    "from epitator.date_annotator import DateAnnotator\r\n",
    "from epitator.geoname_annotator import GeonameAnnotator"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "source": [
    "from transformers import BartForConditionalGeneration, BartTokenizer\r\n",
    "from typing import List\r\n",
    "\r\n",
    "# setup our BART transformer summarization model\r\n",
    "tokenizer = BartTokenizer.from_pretrained('facebook/bart-large-cnn')\r\n",
    "model = BartForConditionalGeneration.from_pretrained('facebook/bart-large-cnn').cuda()"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "source": [
    "# helper function to summarize an input text with the BART model\r\n",
    "def summarizer(text: str) -> str:\r\n",
    "    input_ids = tokenizer(text, return_tensors='pt', max_length=1024, padding=True, truncation=True)['input_ids']\r\n",
    "    summary_ids = model.generate(input_ids.cuda())\r\n",
    "    summary = ''.join([tokenizer.decode(s) for s in summary_ids])\r\n",
    "    summary = summary.replace('<s>', '').replace('</s>', '')\r\n",
    "    return summary"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "source": [
    "import time"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "source": [
    "# function that extracts location names/admin codes/lat/lng, case and death counts, and date ranges from the input string\r\n",
    "# uses epitator since it already trained rules for extracting medical/infectious disease data\r\n",
    "def epitator_extract(txt, max_ents=1):\r\n",
    "    # input string and add annotators\r\n",
    "    t0 = time.time()\r\n",
    "\r\n",
    "    doc = AnnoDoc(txt)\r\n",
    "    print('doc', time.time()-t0)\r\n",
    "    doc.add_tiers(GeonameAnnotator())\r\n",
    "    print('geoname', time.time()-t0)\r\n",
    "    doc.add_tiers(CountAnnotator())\r\n",
    "    print('count', time.time()-t0)\r\n",
    "    doc.add_tiers(DateAnnotator())\r\n",
    "    print('date', time.time()-t0)\r\n",
    "\r\n",
    "    # extract geographic data\r\n",
    "    geos = doc.tiers[\"geonames\"].spans\r\n",
    "    geo_admin1s = [x.geoname.admin1_code for x in geos]\r\n",
    "    geo_admin2s = [x.geoname.admin2_code for x in geos]\r\n",
    "    geo_admin3s = [x.geoname.admin3_code for x in geos]\r\n",
    "    geo_admin4s = [x.geoname.admin4_code for x in geos]\r\n",
    "    geo_names = [x.geoname.name for x in geos]\r\n",
    "    geo_lats = [x.geoname.latitude for x in geos]\r\n",
    "    geo_lons = [x.geoname.longitude for x in geos]\r\n",
    "\r\n",
    "    # extract case counts and death counts\r\n",
    "    counts = doc.tiers[\"counts\"].spans\r\n",
    "    cases_counts = [x.metadata['count'] for x in counts if 'case' in x.metadata['attributes'] and 'death' not in x.metadata['attributes']]\r\n",
    "    cases_tags = [x.metadata['attributes'] for x in counts if 'case' in x.metadata['attributes'] and 'death' not in x.metadata['attributes']]\r\n",
    "    death_counts = [x.metadata['count'] for x in counts if 'death' in x.metadata['attributes']]\r\n",
    "    death_tags = [x.metadata['attributes'] for x in counts if 'death' in x.metadata['attributes']]\r\n",
    "\r\n",
    "    # extract the date range\r\n",
    "    dates = doc.tiers[\"dates\"].spans\r\n",
    "    dates_start = [pd.to_datetime(x.metadata[\"datetime_range\"][0], errors='coerce') for x in dates]\r\n",
    "    dates_end = [pd.to_datetime(x.metadata[\"datetime_range\"][1], errors='coerce') for x in dates]\r\n",
    "\r\n",
    "    # return only max_ents entities from the extracted lists\r\n",
    "    # currently set to the first result for each list, since that is usually the most important one\r\n",
    "    # and other ones can be filler/garbage data\r\n",
    "    return pd.Series([ \r\n",
    "        geo_admin1s[:max_ents],\r\n",
    "        geo_admin2s[:max_ents],\r\n",
    "        geo_admin3s[:max_ents],\r\n",
    "        geo_admin4s[:max_ents],\r\n",
    "        geo_names[:max_ents],\r\n",
    "        geo_lats[:max_ents],\r\n",
    "        geo_lons[:max_ents],\r\n",
    "        cases_counts[:max_ents],\r\n",
    "        cases_tags[:max_ents],\r\n",
    "        death_counts[:max_ents],\r\n",
    "        death_tags[:max_ents],\r\n",
    "        dates_start[:max_ents],\r\n",
    "        dates_end[:max_ents],\r\n",
    "    ])"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "source": [
    "df['summary'] = df['content'].progress_apply(summarizer)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      " 12%|█▏        | 90/743 [01:43<12:32,  1.15s/it]\n"
     ]
    },
    {
     "output_type": "error",
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-134-4f8d5b1975a8>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'summary'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'content'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mprogress_apply\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msummarizer\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\tqdm\\std.py\u001b[0m in \u001b[0;36minner\u001b[1;34m(df, func, *args, **kwargs)\u001b[0m\n\u001b[0;32m    815\u001b[0m                 \u001b[1;31m# on the df using our wrapper (which provides bar updating)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    816\u001b[0m                 \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 817\u001b[1;33m                     \u001b[1;32mreturn\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdf_function\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mwrapper\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    818\u001b[0m                 \u001b[1;32mfinally\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    819\u001b[0m                     \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\pandas\\core\\series.py\u001b[0m in \u001b[0;36mapply\u001b[1;34m(self, func, convert_dtype, args, **kwds)\u001b[0m\n\u001b[0;32m   4136\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4137\u001b[0m                 \u001b[0mvalues\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobject\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 4138\u001b[1;33m                 \u001b[0mmapped\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mlib\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmap_infer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mconvert\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mconvert_dtype\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   4139\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4140\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmapped\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmapped\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mSeries\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\lib.pyx\u001b[0m in \u001b[0;36mpandas._libs.lib.map_infer\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\tqdm\\std.py\u001b[0m in \u001b[0;36mwrapper\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    810\u001b[0m                     \u001b[1;31m# take a fast or slow code path; so stop when t.total==t.n\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    811\u001b[0m                     \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mn\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1\u001b[0m \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtotal\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mn\u001b[0m \u001b[1;33m<\u001b[0m \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtotal\u001b[0m \u001b[1;32melse\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 812\u001b[1;33m                     \u001b[1;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    813\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    814\u001b[0m                 \u001b[1;31m# Apply the provided function (in **kwargs)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-132-85c078e339b3>\u001b[0m in \u001b[0;36msummarizer\u001b[1;34m(text)\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[1;32mdef\u001b[0m \u001b[0msummarizer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m     \u001b[0minput_ids\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtokenizer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mreturn_tensors\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'pt'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmax_length\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1024\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpadding\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtruncation\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'input_ids'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 4\u001b[1;33m     \u001b[0msummary_ids\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmodel\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgenerate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0minput_ids\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcuda\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      5\u001b[0m     \u001b[0msummary\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m''\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mtokenizer\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdecode\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ms\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0ms\u001b[0m \u001b[1;32min\u001b[0m \u001b[0msummary_ids\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      6\u001b[0m     \u001b[0msummary\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msummary\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'<s>'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m''\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'</s>'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m''\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\torch\\autograd\\grad_mode.py\u001b[0m in \u001b[0;36mdecorate_context\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m     26\u001b[0m         \u001b[1;32mdef\u001b[0m \u001b[0mdecorate_context\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     27\u001b[0m             \u001b[1;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__class__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 28\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     29\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mcast\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mF\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdecorate_context\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     30\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\transformers\\generation_utils.py\u001b[0m in \u001b[0;36mgenerate\u001b[1;34m(self, input_ids, max_length, min_length, do_sample, early_stopping, num_beams, temperature, top_k, top_p, repetition_penalty, bad_words_ids, bos_token_id, pad_token_id, eos_token_id, length_penalty, no_repeat_ngram_size, encoder_no_repeat_ngram_size, num_return_sequences, max_time, max_new_tokens, decoder_start_token_id, use_cache, num_beam_groups, diversity_penalty, prefix_allowed_tokens_fn, output_attentions, output_hidden_states, output_scores, return_dict_in_generate, forced_bos_token_id, forced_eos_token_id, remove_invalid_values, synced_gpus, **model_kwargs)\u001b[0m\n\u001b[0;32m   1051\u001b[0m                 \u001b[0minput_ids\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mexpand_size\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mnum_beams\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mis_encoder_decoder\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mconfig\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_encoder_decoder\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mmodel_kwargs\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1052\u001b[0m             )\n\u001b[1;32m-> 1053\u001b[1;33m             return self.beam_search(\n\u001b[0m\u001b[0;32m   1054\u001b[0m                 \u001b[0minput_ids\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1055\u001b[0m                 \u001b[0mbeam_scorer\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\transformers\\generation_utils.py\u001b[0m in \u001b[0;36mbeam_search\u001b[1;34m(self, input_ids, beam_scorer, logits_processor, stopping_criteria, max_length, pad_token_id, eos_token_id, output_attentions, output_hidden_states, output_scores, return_dict_in_generate, synced_gpus, **model_kwargs)\u001b[0m\n\u001b[0;32m   1788\u001b[0m             \u001b[0mmodel_inputs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mprepare_inputs_for_generation\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0minput_ids\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mmodel_kwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1789\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1790\u001b[1;33m             outputs = self(\n\u001b[0m\u001b[0;32m   1791\u001b[0m                 \u001b[1;33m**\u001b[0m\u001b[0mmodel_inputs\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1792\u001b[0m                 \u001b[0mreturn_dict\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\torch\\nn\\modules\\module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[1;34m(self, *input, **kwargs)\u001b[0m\n\u001b[0;32m   1049\u001b[0m         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks\n\u001b[0;32m   1050\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[1;32m-> 1051\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1052\u001b[0m         \u001b[1;31m# Do not call functions when jit is used\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1053\u001b[0m         \u001b[0mfull_backward_hooks\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnon_full_backward_hooks\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\transformers\\models\\bart\\modeling_bart.py\u001b[0m in \u001b[0;36mforward\u001b[1;34m(self, input_ids, attention_mask, decoder_input_ids, decoder_attention_mask, head_mask, decoder_head_mask, cross_attn_head_mask, encoder_outputs, past_key_values, inputs_embeds, decoder_inputs_embeds, labels, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[0;32m   1291\u001b[0m                 )\n\u001b[0;32m   1292\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1293\u001b[1;33m         outputs = self.model(\n\u001b[0m\u001b[0;32m   1294\u001b[0m             \u001b[0minput_ids\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1295\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mattention_mask\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\torch\\nn\\modules\\module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[1;34m(self, *input, **kwargs)\u001b[0m\n\u001b[0;32m   1049\u001b[0m         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks\n\u001b[0;32m   1050\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[1;32m-> 1051\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1052\u001b[0m         \u001b[1;31m# Do not call functions when jit is used\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1053\u001b[0m         \u001b[0mfull_backward_hooks\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnon_full_backward_hooks\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\transformers\\models\\bart\\modeling_bart.py\u001b[0m in \u001b[0;36mforward\u001b[1;34m(self, input_ids, attention_mask, decoder_input_ids, decoder_attention_mask, head_mask, decoder_head_mask, cross_attn_head_mask, encoder_outputs, past_key_values, inputs_embeds, decoder_inputs_embeds, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[0;32m   1182\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1183\u001b[0m         \u001b[1;31m# decoder outputs consists of (dec_features, past_key_value, dec_hidden, dec_attn)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1184\u001b[1;33m         decoder_outputs = self.decoder(\n\u001b[0m\u001b[0;32m   1185\u001b[0m             \u001b[0minput_ids\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdecoder_input_ids\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1186\u001b[0m             \u001b[0mattention_mask\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdecoder_attention_mask\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\torch\\nn\\modules\\module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[1;34m(self, *input, **kwargs)\u001b[0m\n\u001b[0;32m   1049\u001b[0m         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks\n\u001b[0;32m   1050\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[1;32m-> 1051\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1052\u001b[0m         \u001b[1;31m# Do not call functions when jit is used\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1053\u001b[0m         \u001b[0mfull_backward_hooks\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnon_full_backward_hooks\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\transformers\\models\\bart\\modeling_bart.py\u001b[0m in \u001b[0;36mforward\u001b[1;34m(self, input_ids, attention_mask, encoder_hidden_states, encoder_attention_mask, head_mask, cross_attn_head_mask, past_key_values, inputs_embeds, use_cache, output_attentions, output_hidden_states, return_dict)\u001b[0m\n\u001b[0;32m   1049\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1050\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1051\u001b[1;33m                 layer_outputs = decoder_layer(\n\u001b[0m\u001b[0;32m   1052\u001b[0m                     \u001b[0mhidden_states\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1053\u001b[0m                     \u001b[0mattention_mask\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mattention_mask\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\torch\\nn\\modules\\module.py\u001b[0m in \u001b[0;36m_call_impl\u001b[1;34m(self, *input, **kwargs)\u001b[0m\n\u001b[0;32m   1049\u001b[0m         if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks\n\u001b[0;32m   1050\u001b[0m                 or _global_forward_hooks or _global_forward_pre_hooks):\n\u001b[1;32m-> 1051\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mforward_call\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1052\u001b[0m         \u001b[1;31m# Do not call functions when jit is used\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1053\u001b[0m         \u001b[0mfull_backward_hooks\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnon_full_backward_hooks\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\transformers\\models\\bart\\modeling_bart.py\u001b[0m in \u001b[0;36mforward\u001b[1;34m(self, hidden_states, attention_mask, encoder_hidden_states, encoder_attention_mask, layer_head_mask, cross_attn_layer_head_mask, past_key_value, output_attentions, use_cache)\u001b[0m\n\u001b[0;32m    427\u001b[0m         \u001b[1;31m# Fully Connected\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    428\u001b[0m         \u001b[0mresidual\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mhidden_states\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 429\u001b[1;33m         \u001b[0mhidden_states\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mactivation_fn\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfc1\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhidden_states\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    430\u001b[0m         \u001b[0mhidden_states\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mnn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfunctional\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdropout\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhidden_states\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mp\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mactivation_dropout\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtraining\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtraining\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    431\u001b[0m         \u001b[0mhidden_states\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfc2\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhidden_states\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\torch\\nn\\functional.py\u001b[0m in \u001b[0;36mgelu\u001b[1;34m(input)\u001b[0m\n\u001b[0;32m   1553\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mhas_torch_function_unary\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1554\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mhandle_torch_function\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgelu\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0minput\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1555\u001b[1;33m     \u001b[1;32mreturn\u001b[0m \u001b[0mtorch\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_C\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_nn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgelu\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1556\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1557\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "df[['admin1_code',\r\n",
    "'admin2_code',\r\n",
    "'admin3_code',\r\n",
    "'admin4_code',\r\n",
    "'location_name',\r\n",
    "'location_lat',\r\n",
    "'location_lon',\r\n",
    "'cases',\r\n",
    "'cases_tags',\r\n",
    "'deaths',\r\n",
    "'deaths_tags',\r\n",
    "'dates_start',\r\n",
    "'dates_end',]] = df['summary'].progress_apply(epitator_extract)\r\n",
    "df = df.applymap(lambda x: x[0] if isinstance(x, list) and len(x) > 0 else x)\r\n",
    "df = df.applymap(lambda y: pd.NA if isinstance(y, (list, str)) and len(y) == 0 else y)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "100%|██████████| 743/743 [01:02<00:00, 11.89it/s]\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "df.iloc[0]"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "Unnamed: 0                                                          0\n",
       "id                                                            8531354\n",
       "title                                       ['Malaria - Yemen: (TA)']\n",
       "zoom_lat                                                    13.566667\n",
       "zoom_lon                                                    44.033333\n",
       "zoom_level                                                          4\n",
       "alert_id                                                      8531354\n",
       "feed_id                                                             1\n",
       "summary             Health facilities in the province have recorde...\n",
       "issue_date                                        2021-07-20 14:42:19\n",
       "load_date                                         2021-07-20 14:42:19\n",
       "incident_date                                                     NaN\n",
       "descr               Thousands of citizens have been infected with ...\n",
       "alert_tag_id                                                      NaN\n",
       "dup_count                                                           0\n",
       "dup_of                                                            NaN\n",
       "unique_string                                         20210720.853135\n",
       "info_hash                                                         NaN\n",
       "submitted_by                                                   3408.0\n",
       "reviewed                                                          NaN\n",
       "search_string_id                                                  NaN\n",
       "content             Thousands of citizens have been infected with ...\n",
       "admin1_code                                                        08\n",
       "admin2_code                                                      <NA>\n",
       "admin3_code                                                      <NA>\n",
       "admin4_code                                                      <NA>\n",
       "location_name                              State of Western Australia\n",
       "location_lat                                                    -25.0\n",
       "location_lon                                                    122.0\n",
       "cases                                                            4000\n",
       "cases_tags                                          [case, suspected]\n",
       "deaths                                                           <NA>\n",
       "deaths_tags                                                      <NA>\n",
       "dates_start                                                      <NA>\n",
       "dates_end                                                        <NA>\n",
       "Name: 0, dtype: object"
      ]
     },
     "metadata": {},
     "execution_count": 27
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "df.to_csv('promed_malaria_parsed.csv', sep='\\t')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "len(df)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "743"
      ]
     },
     "metadata": {},
     "execution_count": 29
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "data = 'GPM_3IMERGHH_06_precipitationCal'\r\n",
    "lat = '13.566667'\r\n",
    "lon = '44.033333'\r\n",
    "time_start = '2021-07-20T14:42:19'\r\n",
    "time_end = '2021-07-20T14:42:19'"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "source": [
    "df.iloc[1][['zoom_lat', 'zoom_lon', 'issue_date']]"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "zoom_lat                36.519981\n",
       "zoom_lon               103.891769\n",
       "issue_date    2021-07-06 10:44:55\n",
       "Name: 1, dtype: object"
      ]
     },
     "metadata": {},
     "execution_count": 173
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "source": [
    "import requests"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "f'https://api.giovanni.earthdata.nasa.gov/timeseries?data=${data}&location=%5B${lat}%2C${lon}%5D&time=${time_start}%2F${time_end}'"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'https://api.giovanni.earthdata.nasa.gov/timeseries?data=$GPM_3IMERGHH_06_precipitationCal&location=%5B$13.566667%2C$44.033333%5D&time=$2021-07-20T14:42:19%2F$2021-07-20T14:42:19'"
      ]
     },
     "metadata": {},
     "execution_count": 52
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "from datetime import datetime\r\n",
    "from dateutil.relativedelta import relativedelta"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "source": [
    "def get_gpm(row):\r\n",
    "    headers = {\r\n",
    "        'authorizationtoken': 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoieWVlYiIsInRva2VuIjoiZXlKMGVYQWlPaUpLVjFRaUxDSnZjbWxuYVc0aU9pSkZZWEowYUdSaGRHRWdURzluYVc0aUxDSmhiR2NpT2lKSVV6STFOaUo5LmV5SjBlWEJsSWpvaVQwRjFkR2dpTENKMWFXUWlPaUo1WldWaUlpd2lZMnhwWlc1MFgybGtJam9pWlRKWFZtczRVSGMyZDJWbFRGVkxXbGxQZUhaVVVTSXNJbVY0Y0NJNk1UWXlPVGt4TVRreU9Dd2lhV0YwSWpveE5qSTNNekU1T1RJNExDSnBjM01pT2lKRllYSjBhR1JoZEdFZ1RHOW5hVzRpZlEuYk9lcVlkQ2UzU0lOVWpsNWNadmtSYU5LU0RvSkN4VWRlRmZMT0tLNHFOYyIsImlhdCI6MTYyNzMxOTkyOH0.O7A7WYds19vi0R3q4THfBHHHXeqYWPJca2djiBsqwGA',\r\n",
    "    }\r\n",
    "\r\n",
    "    time = datetime.strptime(row['issue_date'], '%Y-%m-%d %H:%M:%S')\r\n",
    "    time_start = time.strftime('%Y-%m-%dT00:00:00')\r\n",
    "    time_end = (time + relativedelta(months=1)).strftime('%Y-%m-%dT00:00:00')\r\n",
    "\r\n",
    "    params = (\r\n",
    "        ('data', 'GPM_3IMERGHH_06_precipitationCal'),\r\n",
    "        ('location', f'[{row[\"zoom_lat\"]},{row[\"zoom_lon\"]}]'),\r\n",
    "        ('time', f'{time_start}/{time_end}'),\r\n",
    "    )\r\n",
    "    print(params)\r\n",
    "\r\n",
    "    r = requests.get('https://api.giovanni.earthdata.nasa.gov/timeseries', headers=headers, params=params)\r\n",
    "    print(r.text)\r\n",
    "    if 'mean,' in r.text:\r\n",
    "        mean_idx = r.text.find('mean,')\r\n",
    "        return r.text[mean_idx+5:mean_idx+r.text[mean_idx:].find('\\n')]\r\n",
    "    # print('uhoh')\r\n",
    "    return pd.NA\r\n"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "source": [
    "get_gpm(df.iloc[60])"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "(('data', 'GPM_3IMERGHH_06_precipitationCal'), ('location', '[75.0,-39.990234]'), ('time', '2018-07-23T00:00:00/2018-08-23T00:00:00'))\n",
      "{\"Message\":\"User is not authorized to access this resource with an explicit deny\"}\n"
     ]
    },
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "<NA>"
      ]
     },
     "metadata": {},
     "execution_count": 197
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "source": [
    "df['gpm'] = df.progress_apply(get_gpm, axis=1)\r\n",
    "df.to_csv('promed_malaria_gpm.csv', sep='\\t')"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "  2%|▏         | 13/743 [00:10<10:15,  1.19it/s]\n"
     ]
    },
    {
     "output_type": "error",
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-181-4eda51777c65>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'gpm'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mprogress_apply\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mget_gpm\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'promed_malaria_gpm.csv'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msep\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'\\t'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\tqdm\\std.py\u001b[0m in \u001b[0;36minner\u001b[1;34m(df, func, *args, **kwargs)\u001b[0m\n\u001b[0;32m    815\u001b[0m                 \u001b[1;31m# on the df using our wrapper (which provides bar updating)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    816\u001b[0m                 \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 817\u001b[1;33m                     \u001b[1;32mreturn\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdf_function\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mwrapper\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    818\u001b[0m                 \u001b[1;32mfinally\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    819\u001b[0m                     \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\pandas\\core\\frame.py\u001b[0m in \u001b[0;36mapply\u001b[1;34m(self, func, axis, raw, result_type, args, **kwds)\u001b[0m\n\u001b[0;32m   7766\u001b[0m             \u001b[0mkwds\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mkwds\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   7767\u001b[0m         )\n\u001b[1;32m-> 7768\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   7769\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   7770\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mapplymap\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mna_action\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mOptional\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mstr\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[0mDataFrame\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\pandas\\core\\apply.py\u001b[0m in \u001b[0;36mget_result\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    183\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mapply_raw\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    184\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 185\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mapply_standard\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    186\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    187\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mapply_empty_result\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\pandas\\core\\apply.py\u001b[0m in \u001b[0;36mapply_standard\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    274\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    275\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mapply_standard\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 276\u001b[1;33m         \u001b[0mresults\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mres_index\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mapply_series_generator\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    277\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    278\u001b[0m         \u001b[1;31m# wrap results\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\pandas\\core\\apply.py\u001b[0m in \u001b[0;36mapply_series_generator\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    288\u001b[0m             \u001b[1;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mv\u001b[0m \u001b[1;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mseries_gen\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    289\u001b[0m                 \u001b[1;31m# ignore SettingWithCopy here in case the user mutates\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 290\u001b[1;33m                 \u001b[0mresults\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mf\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mv\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    291\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mresults\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mABCSeries\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    292\u001b[0m                     \u001b[1;31m# If we have a view on v, we need to make a copy because\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\tqdm\\std.py\u001b[0m in \u001b[0;36mwrapper\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m    810\u001b[0m                     \u001b[1;31m# take a fast or slow code path; so stop when t.total==t.n\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    811\u001b[0m                     \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mn\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1\u001b[0m \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtotal\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mn\u001b[0m \u001b[1;33m<\u001b[0m \u001b[0mt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtotal\u001b[0m \u001b[1;32melse\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 812\u001b[1;33m                     \u001b[1;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    813\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    814\u001b[0m                 \u001b[1;31m# Apply the provided function (in **kwargs)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<ipython-input-180-ca2fea29351c>\u001b[0m in \u001b[0;36mget_gpm\u001b[1;34m(row)\u001b[0m\n\u001b[0;32m     15\u001b[0m     )\n\u001b[0;32m     16\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 17\u001b[1;33m     \u001b[0mr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'https://api.giovanni.earthdata.nasa.gov/timeseries'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mheaders\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     18\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[1;34m'mean,'\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     19\u001b[0m         \u001b[0mmean_idx\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfind\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'mean,'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\requests\\api.py\u001b[0m in \u001b[0;36mget\u001b[1;34m(url, params, **kwargs)\u001b[0m\n\u001b[0;32m     74\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     75\u001b[0m     \u001b[0mkwargs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msetdefault\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'allow_redirects'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 76\u001b[1;33m     \u001b[1;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'get'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     77\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     78\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\requests\\api.py\u001b[0m in \u001b[0;36mrequest\u001b[1;34m(method, url, **kwargs)\u001b[0m\n\u001b[0;32m     59\u001b[0m     \u001b[1;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     60\u001b[0m     \u001b[1;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 61\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     62\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     63\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\requests\\sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[0;32m    540\u001b[0m         }\n\u001b[0;32m    541\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 542\u001b[1;33m         \u001b[0mresp\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    543\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    544\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\requests\\sessions.py\u001b[0m in \u001b[0;36msend\u001b[1;34m(self, request, **kwargs)\u001b[0m\n\u001b[0;32m    653\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    654\u001b[0m         \u001b[1;31m# Send the request\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 655\u001b[1;33m         \u001b[0mr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    656\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    657\u001b[0m         \u001b[1;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\requests\\adapters.py\u001b[0m in \u001b[0;36msend\u001b[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[0;32m    437\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    438\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mchunked\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 439\u001b[1;33m                 resp = conn.urlopen(\n\u001b[0m\u001b[0;32m    440\u001b[0m                     \u001b[0mmethod\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    441\u001b[0m                     \u001b[0murl\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    697\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    698\u001b[0m             \u001b[1;31m# Make the request on the httplib connection object.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 699\u001b[1;33m             httplib_response = self._make_request(\n\u001b[0m\u001b[0;32m    700\u001b[0m                 \u001b[0mconn\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    701\u001b[0m                 \u001b[0mmethod\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    443\u001b[0m                     \u001b[1;31m# Python 3 (including for exceptions like SystemExit).\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    444\u001b[0m                     \u001b[1;31m# Otherwise it looks like a bug in the code.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 445\u001b[1;33m                     \u001b[0msix\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mraise_from\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    446\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mSocketError\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    447\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_raise_timeout\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0merr\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0me\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0murl\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtimeout_value\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mread_timeout\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\urllib3\\packages\\six.py\u001b[0m in \u001b[0;36mraise_from\u001b[1;34m(value, from_value)\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\site-packages\\urllib3\\connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    438\u001b[0m                 \u001b[1;31m# Python 3\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    439\u001b[0m                 \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 440\u001b[1;33m                     \u001b[0mhttplib_response\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mconn\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgetresponse\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    441\u001b[0m                 \u001b[1;32mexcept\u001b[0m \u001b[0mBaseException\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    442\u001b[0m                     \u001b[1;31m# Remove the TypeError from the exception chain in\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\http\\client.py\u001b[0m in \u001b[0;36mgetresponse\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m   1342\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1343\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1344\u001b[1;33m                 \u001b[0mresponse\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbegin\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1345\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mConnectionError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1346\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\http\\client.py\u001b[0m in \u001b[0;36mbegin\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    305\u001b[0m         \u001b[1;31m# read until we get a non-100 response\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    306\u001b[0m         \u001b[1;32mwhile\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 307\u001b[1;33m             \u001b[0mversion\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mstatus\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mreason\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_read_status\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    308\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mstatus\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[0mCONTINUE\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    309\u001b[0m                 \u001b[1;32mbreak\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\http\\client.py\u001b[0m in \u001b[0;36m_read_status\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    266\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    267\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_read_status\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 268\u001b[1;33m         \u001b[0mline\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreadline\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0m_MAXLINE\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;36m1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"iso-8859-1\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    269\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mline\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m>\u001b[0m \u001b[0m_MAXLINE\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    270\u001b[0m             \u001b[1;32mraise\u001b[0m \u001b[0mLineTooLong\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"status line\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\socket.py\u001b[0m in \u001b[0;36mreadinto\u001b[1;34m(self, b)\u001b[0m\n\u001b[0;32m    667\u001b[0m         \u001b[1;32mwhile\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    668\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 669\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mb\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    670\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    671\u001b[0m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_timeout_occurred\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\ssl.py\u001b[0m in \u001b[0;36mrecv_into\u001b[1;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[0;32m   1239\u001b[0m                   \u001b[1;34m\"non-zero flags not allowed in calls to recv_into() on %s\"\u001b[0m \u001b[1;33m%\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1240\u001b[0m                   self.__class__)\n\u001b[1;32m-> 1241\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnbytes\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1242\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1243\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0msuper\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mbuffer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnbytes\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mF:\\conda\\envs\\promed2021\\lib\\ssl.py\u001b[0m in \u001b[0;36mread\u001b[1;34m(self, len, buffer)\u001b[0m\n\u001b[0;32m   1097\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1098\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mbuffer\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1099\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1100\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1101\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "source": [
    "headers = {\r\n",
    "    'authorizationtoken': 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoieWVlYiIsInRva2VuIjoiZXlKMGVYQWlPaUpLVjFRaUxDSnZjbWxuYVc0aU9pSkZZWEowYUdSaGRHRWdURzluYVc0aUxDSmhiR2NpT2lKSVV6STFOaUo5LmV5SjBlWEJsSWpvaVQwRjFkR2dpTENKMWFXUWlPaUo1WldWaUlpd2lZMnhwWlc1MFgybGtJam9pWlRKWFZtczRVSGMyZDJWbFRGVkxXbGxQZUhaVVVTSXNJbVY0Y0NJNk1UWXpNVEE0TURBd09Td2lhV0YwSWpveE5qSTRORGc0TURBNUxDSnBjM01pT2lKRllYSjBhR1JoZEdFZ1RHOW5hVzRpZlEuZ3RXM1MzdjVQZlNUdk1iaHZiZXJjcV85b2R5WWxpRV84ZGMxM3lxMmVjWSIsImlhdCI6MTYyODQ4ODAwOX0.BvWgKN1ZTpqbs4uHFYU3dLBavtCY2bRvyqf-NVF8AfM',\r\n",
    "}\r\n",
    "# df.iloc[1]['zoom_lat'], 'zoom_lon', 'issue_date']]\r\n",
    "# params = (\r\n",
    "#     ('data', 'GPM_3IMERGHH_06_precipitationCal'),\r\n",
    "#     ('location', f'[{df.iloc[1][\"zoom_lat\"]},{df.iloc[1][\"zoom_lon\"]}]'),\r\n",
    "#     ('time', '2018-07-23T00:00:00/2018-08-23T00:00:00'),\r\n",
    "# )\r\n",
    "params = (('data', 'GPM_3IMERGHH_06_precipitationCal'), ('location',\r\n",
    "          '[75.0,-39.990234]'), ('time', '2018-07-23T00:00:00/2018-08-23T00:00:00'))\r\n",
    "\r\n",
    "response = requests.get(\r\n",
    "    'https://api.giovanni.earthdata.nasa.gov/timeseries', headers=headers, params=params)\r\n"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "source": [
    "print(response.text)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "{\"Message\":\"User is not authorized to access this resource with an explicit deny\"}\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "source": [
    "print(response)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "<Response [403]>\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "source": [
    "headers = {\r\n",
    "    'authorizationtoken': 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoieWVlYiIsInRva2VuIjoiZXlKMGVYQWlPaUpLVjFRaUxDSnZjbWxuYVc0aU9pSkZZWEowYUdSaGRHRWdURzluYVc0aUxDSmhiR2NpT2lKSVV6STFOaUo5LmV5SjBlWEJsSWpvaVQwRjFkR2dpTENKMWFXUWlPaUo1WldWaUlpd2lZMnhwWlc1MFgybGtJam9pWlRKWFZtczRVSGMyZDJWbFRGVkxXbGxQZUhaVVVTSXNJbVY0Y0NJNk1UWXlPVGcyTnprNU9Dd2lhV0YwSWpveE5qSTNNamMxT1RrNExDSnBjM01pT2lKRllYSjBhR1JoZEdFZ1RHOW5hVzRpZlEucmFkZFl0ZHRuOXlGQ3RwOTNENHRselp3bVU1Y1ZVRjFoclJYZXY5dlhQMCIsImlhdCI6MTYyNzI3NTk5OX0.0y_IMp4_2cUrUWzZRVmkL5MBdDPI7VzFLPtAQb1_fEk',\r\n",
    "}\r\n",
    "\r\n",
    "params = (\r\n",
    "    ('data', 'GPM_3IMERGHH_06_precipitationCal'),\r\n",
    "    ('location', '[4.75,0.55]'),\r\n",
    "    ('time', '2000-06-01T00:00:00/2000-06-01T07:30:00'),\r\n",
    ")\r\n",
    "\r\n",
    "response = requests.get('https://api.giovanni.earthdata.nasa.gov/timeseries', headers=headers, params=params)"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "source": [
    "print(response)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "<Response [403]>\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "source": [
    "curl -X GET \"https://api.giovanni.earthdata.nasa.gov/timeseries?data=GPM_3IMERGHH_06_precipitationCal&location=%5B4.75%2C0.55%5D&time=2000-06-01T00:00:00%2F2000-06-01T07:30:00\" -H \"authorizationtoken: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoieWVlYiIsInRva2VuIjoiZXlKMGVYQWlPaUpLVjFRaUxDSnZjbWxuYVc0aU9pSkZZWEowYUdSaGRHRWdURzluYVc0aUxDSmhiR2NpT2lKSVV6STFOaUo5LmV5SjBlWEJsSWpvaVQwRjFkR2dpTENKMWFXUWlPaUo1WldWaUlpd2lZMnhwWlc1MFgybGtJam9pWlRKWFZtczRVSGMyZDJWbFRGVkxXbGxQZUhaVVVTSXNJbVY0Y0NJNk1UWXlPVGt4TVRreU9Dd2lhV0YwSWpveE5qSTNNekU1T1RJNExDSnBjM01pT2lKRllYSjBhR1JoZEdFZ1RHOW5hVzRpZlEuYk9lcVlkQ2UzU0lOVWpsNWNadmtSYU5LU0RvSkN4VWRlRmZMT0tLNHFOYyIsImlhdCI6MTYyNzMxOTkyOH0.O7A7WYds19vi0R3q4THfBHHHXeqYWPJca2djiBsqwGA\""
   ],
   "outputs": [
    {
     "output_type": "error",
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-198-bc5436a29e9f>, line 1)",
     "traceback": [
      "\u001b[1;36m  File \u001b[1;32m\"<ipython-input-198-bc5436a29e9f>\"\u001b[1;36m, line \u001b[1;32m1\u001b[0m\n\u001b[1;33m    curl -X GET \"https://api.giovanni.earthdata.nasa.gov/timeseries?data=GPM_3IMERGHH_06_precipitationCal&location=%5B4.75%2C0.55%5D&time=2000-06-01T00:00:00%2F2000-06-01T07:30:00\" -H \"authorizationtoken: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoieWVlYiIsInRva2VuIjoiZXlKMGVYQWlPaUpLVjFRaUxDSnZjbWxuYVc0aU9pSkZZWEowYUdSaGRHRWdURzluYVc0aUxDSmhiR2NpT2lKSVV6STFOaUo5LmV5SjBlWEJsSWpvaVQwRjFkR2dpTENKMWFXUWlPaUo1WldWaUlpd2lZMnhwWlc1MFgybGtJam9pWlRKWFZtczRVSGMyZDJWbFRGVkxXbGxQZUhaVVVTSXNJbVY0Y0NJNk1UWXlPVGt4TVRreU9Dd2lhV0YwSWpveE5qSTNNekU1T1RJNExDSnBjM01pT2lKRllYSjBhR1JoZEdFZ1RHOW5hVzRpZlEuYk9lcVlkQ2UzU0lOVWpsNWNadmtSYU5LU0RvSkN4VWRlRmZMT0tLNHFOYyIsImlhdCI6MTYyNzMxOTkyOH0.O7A7WYds19vi0R3q4THfBHHHXeqYWPJca2djiBsqwGA\"\u001b[0m\n\u001b[1;37m            ^\u001b[0m\n\u001b[1;31mSyntaxError\u001b[0m\u001b[1;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "source": [
    "import pandas as pd"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "source": [
    "df = pd.read_feather('dataset.v1.0.feather')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "source": [
    "content = df.loc[(df['disease'] == 'malaria')& (df['cases'].isna())][['content', 'summary']]"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "source": [
    "content.iloc[6]['content']"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'Citation: Uwimana A, Umulisa N, Venkatesan M, et al. Association of _Plasmodium falciparum kelch13_ R561H genotypes with delayed parasite clearance in Rwanda: an open-label, single-arm, multicentre, therapeutic efficacy study. Lancet. Epub 14 Apr 2021. <https://doi.org/10.1016/S1473-3099(21)00142-0>\\nSummary\\n------\\nBackground: Partial artemisinin resistance is suspected if delayed parasite clearance (i.e., persistence of parasitaemia on day 3 after treatment initiation) is observed. Validated markers of artemisinin partial resistance in southeast Asia, _Plasmodium falciparum kelch13_ (_Pfkelch13_) R561H and P574L, have been reported in Rwanda, but no association with parasite clearance has been observed. We aimed to establish the efficacy of artemether-lumefantrine and genetic characterisation of _Pfkelch13_ alleles and their association with treatment outcomes.\\nMethods: This open-label, single-arm, multicentre, therapeutic efficacy study was done in 2018 in 3 Rwandan sites: Masaka, Rukara, and Bugarama. Children aged 6-59 months with _P. falciparum_ monoinfection and fever were eligible and treated with a 3-day course of artemether-lumefantrine. Treatment response was monitored for 28 days using weekly microscopy screenings of blood samples for _P. falciparum_. Mutations in _Pfkelch13_ and _P. falciparum_ multidrug resistance-1 (_Pfmdr1_) genes were characterised in parasites collected from enrolled participants. Analysis of flanking microsatellites surrounding _Pfkelch13_ was done to define the origins of the R561H mutations. The primary endpoint was PCR-corrected parasitological cure on day 28, as per WHO protocol.\\nFindings: 228 participants were enrolled and 224 (98.2%) reached the study endpoint. PCR-corrected efficacies were 97.0% (95% CI 88-100) in Masaka, 93.8% (85-98) in Rukara, and 97.2% (91-100) in Bugarama. _Pfkelch13_ R561H mutations were present in 28 (13%) of 218 pre-treatment samples and P574L mutations were present in 2 (1%) pretreatment samples. 217 (90%) of the 240 _Pfmdr1_ haplotypes observed in the pretreatment samples had either the NFD (N86Y, Y184F, D1246Y) or NYD haplotype. Eight (16%) of 51 participants in Masaka and 12 (15%) of 82 participants in Rukara were microscopically positive 3 days after treatment initiation, which was associated with pre-treatment presence of _Pfkelch13_ R561H in Masaka (p = 0.0005). Genetic analysis of _Pfkelch13_ R561H mutations suggests their common ancestry and local origin in Rwanda.\\nInterpretation: We confirm evidence of emerging artemisinin partial resistance in Rwanda. Although artemether-lumefantrine remains efficacious, vigilance for decreasing efficacy, further characterisation of artemisinin partial resistance, and evaluation of additional antimalarials in Rwanda should be considered.'"
      ]
     },
     "metadata": {},
     "execution_count": 15
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "source": [
    "content.iloc[6]['summary']"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'Researchers found the presence of malaria parasite _Plasmodium knowlesi_ in the north Indian population. The zoonotic parasite was found in the states of Uttar Pradesh, Haryana, and Delhi. A previous study at AIIMS had found its presence in the Andaman and Nicobar Islands.'"
      ]
     },
     "metadata": {},
     "execution_count": 17
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "source": [
    "len(df[df['summary'].str.contains('case')])"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "1402"
      ]
     },
     "metadata": {},
     "execution_count": 28
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "source": [
    "len(df)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "2991"
      ]
     },
     "metadata": {},
     "execution_count": 29
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "source": [
    "df[df['summary'].str.contains('case')]['cases'].isna().sum()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "271"
      ]
     },
     "metadata": {},
     "execution_count": 30
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "source": [
    "valid_df = df[(df['summary'].str.contains('case')) & (df['cases'].notna())]"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "source": [
    "valid_df.iloc[0]"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "index                                                                    5\n",
       "Unnamed: 0                                                         7945357\n",
       "id                                                                 7945357\n",
       "title                                 ['Dengue/DHF update (12): Americas']\n",
       "zoom_lat                                                              75.0\n",
       "zoom_lon                                                        -39.990234\n",
       "zoom_level                                                               4\n",
       "alert_id                                                           7945357\n",
       "feed_id                                                                  1\n",
       "summary                  Cases in various countries. WHO/PAHO countries...\n",
       "issue_date                                             2020-11-15 19:55:03\n",
       "load_date                                              2020-11-15 19:55:03\n",
       "incident_date                                                          NaN\n",
       "descr                    Cases by Country / Week updated / Serotype / T...\n",
       "alert_tag_id                                                           NaN\n",
       "dup_count                                                                0\n",
       "dup_of                                                                 NaN\n",
       "unique_string                                              20201115.794536\n",
       "info_hash                                                             None\n",
       "submitted_by                                                        3419.0\n",
       "reviewed                                                              None\n",
       "search_string_id                                                       NaN\n",
       "content                  Mexico \\n---\\nMexico (Veracruz state)\\nCaribbe...\n",
       "disease                                                             dengue\n",
       "gpm_lat_idx                                                           1649\n",
       "gpm_lon_idx                                                           1400\n",
       "precipitation_anomaly                                                  0.0\n",
       "admin1_code                                                             12\n",
       "admin2_code                                                            035\n",
       "admin3_code                                                           None\n",
       "admin4_code                                                           None\n",
       "location_name                                                     Américas\n",
       "location_lat                                                      18.36455\n",
       "location_lon                                                     -99.54467\n",
       "cases                                                               2000.0\n",
       "cases_tags                                               [case, confirmed]\n",
       "deaths                                                              1000.0\n",
       "deaths_tags                                                  [case, death]\n",
       "dates_start                                                            NaT\n",
       "dates_end                                                              NaT\n",
       "Name: 5, dtype: object"
      ]
     },
     "metadata": {},
     "execution_count": 40
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "source": [
    "valid_df['precipitation_anomaly'].corr(valid_df['cases'])"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "-0.013714449294276154"
      ]
     },
     "metadata": {},
     "execution_count": 42
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "source": [
    "summarizer('''Malaria has taken a turn for the worse in tribal dominated remote\r\n",
    "hamlets under Chawmanu rural administrative block of Dhalai district\r\n",
    "in North Tripura with 5 deaths reported. The death toll in diarrhoea,\r\n",
    "viral fever and malaria has reached 129 so far this year in the area.\r\n",
    "\r\n",
    "Officials here today said 7 special medical teams from the district\r\n",
    "headquarters had been dispatched to the affected hamlets with\r\n",
    "necessary medical aids.\r\n",
    "\r\n",
    "Medicine, saline and instant malaria parasite testing kits were sent\r\n",
    "from Agartala while 7 more doctors were posted in various primary\r\n",
    "health centres in the area in the past one month to contain the\r\n",
    "situation, said a top health official. He, however, pointed out that\r\n",
    "scanty rainfall and humid weather were the main reasons for spread of\r\n",
    "the diseases in hilly locations.\r\n",
    "\r\n",
    "The entire Dhalai district had already been identified as malaria\r\n",
    "prone and become drug resistant. Spraying of DDT, use of instant\r\n",
    "diagnostic kits in highly malaria prone areas besides, distribution of\r\n",
    "115 000 medicated nets and fogging machines were on.\r\n",
    "\r\n",
    "Alarmed at the increasing incidence of deaths due to malaria, the\r\n",
    "Tripura National Vector Borne Disease Control Programme (NVBDC) had\r\n",
    "conducted a comprehensive study in malaria prone areas on\r\n",
    "effectiveness of conventional medicines for malaria treatment.\r\n",
    "\r\n",
    "The Central Drug Research Institute (CDRI) categorised the state as\r\n",
    "Chloroquine (malaria drug) resistant 3 years back and Sulpha-\r\n",
    "Pyramethamine Combination, 2nd-generation anti-malaria drugs had been\r\n",
    "prescribed for the patients.\r\n",
    "\r\n",
    "The _Plasmodium falciparum_ (PF) component present in malaria affected\r\n",
    "patients and Chloroquine had become inactive for long-time sufferers. New\r\n",
    "chemoprophylactic drugs had been supplied as an immediate measure, said\r\n",
    "health officials.\r\n",
    "''')"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "F:\\conda\\envs\\geopandas\\lib\\site-packages\\torch\\_tensor.py:575: UserWarning: floor_divide is deprecated, and will be removed in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values.\n",
      "To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor'). (Triggered internally at  ..\\aten\\src\\ATen\\native\\BinaryOps.cpp:467.)\n",
      "  return torch.floor_divide(self, other)\n"
     ]
    },
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'Malaria has taken a turn for the worse in tribal dominated remote hamlets of North Tripura. Death toll in diarrhoea, viral fever and malaria has reached 129 so far this year in the area. 7 special medical teams from the district headquarters have been dispatched to the affected hamlets.'"
      ]
     },
     "metadata": {},
     "execution_count": 45
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "source": [
    "extraction = epitator_extract('''Malaria has claimed the lives of more than 700 people in Gombe state\r\n",
    "in the last 2.5 years according to Mallam Ahmed Audu. Evaluating\r\n",
    "Officer of the World Bank Assisted Malaria Control Booster Project in\r\n",
    "Gombe.\r\n",
    "\r\n",
    "Mallam Ahmed Audu made this revelation recently in Dukku LGA [Local\r\n",
    "Government Area] of the state during an interactive session organized\r\n",
    "for working journalists in the state.\r\n",
    "\r\n",
    "He said more than 100 deaths occurred between February and May this\r\n",
    "year [2011] alone even as he pointed out that about 170 000 people\r\n",
    "were infected by the malaria scourge in the 11 Local Government Areas\r\n",
    "of the State from 2009 to May 2011.\r\n",
    "\r\n",
    "According to the project monitoring officer, about 46 000 out of the\r\n",
    "malaria prevalence rate were recorded between February and May this\r\n",
    "year [2011]. Audu, however, stated that the project had in February\r\n",
    "this year distributed over 130 000 doses of malaria drugs to all\r\n",
    "government hospitals in the 11 local government areas in the state.\r\n",
    "\r\n",
    "He further said about 2 million doses of malaria preventive drugs for\r\n",
    "pregnant women and children were donated to the state Ministry of\r\n",
    "Health for use during the Children Health Week in May this year.\r\n",
    "\r\n",
    "Also speaking, the Project Manager, Dr. Arnold Abel stated that\r\n",
    "malaria accounted for the loss of over N130 billion [about USD 844\r\n",
    "million] annually in Nigeria as well as for 30 percent deaths in\r\n",
    "children and 5 percent in pregnant women.\r\n",
    "\r\n",
    "In his speech, the Project Mobilization officer, Mallam Muhammad\r\n",
    "Lawal, said there were plans to produce jingles, drama, and programmes\r\n",
    "on malaria attacks and treatment on the state radio and television\r\n",
    "stations in an effort to reduce its spread, especially among poor\r\n",
    "rural communities. He also said the Project collaborated with the\r\n",
    "State Environmental Protection Agency (GOSEPA) on refuse disposal\r\n",
    "while plans were underway to resurrect sanitary departments in the 11\r\n",
    "local government areas of the state.\r\n",
    "\r\n",
    "[Byline: Williams Attah]\r\n",
    "''')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "source": [
    "oaa = pd.DataFrame()\r\n",
    "oaa[['admin1_code',\r\n",
    "'admin2_code',\r\n",
    "'admin3_code',\r\n",
    "'admin4_code',\r\n",
    "'location_name',\r\n",
    "'location_lat',\r\n",
    "'location_lon',\r\n",
    "'cases',\r\n",
    "'cases_tags',\r\n",
    "'deaths',\r\n",
    "'deaths_tags',\r\n",
    "'dates_start',\r\n",
    "'dates_end',]] = df['content'].iloc[:6].progress_apply(epitator_extract)"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "  0%|          | 0/6 [00:00<?, ?it/s]"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "doc 0.0\n",
      "geoname 4.67788290977478\n",
      "count 25.025121450424194\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      " 33%|███▎      | 2/6 [00:43<01:27, 21.87s/it]"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "date 43.693984270095825\n",
      "doc 0.0\n",
      "geoname 3.761343240737915\n",
      "count 8.320590734481812\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      " 50%|█████     | 3/6 [00:55<00:52, 17.55s/it]"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "date 11.48423433303833\n",
      "doc 0.0\n",
      "geoname 2.6495070457458496\n",
      "count 9.396372318267822\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      " 67%|██████▋   | 4/6 [01:12<00:35, 17.51s/it]"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "date 17.423551321029663\n",
      "doc 0.0\n",
      "geoname 1.6490018367767334\n",
      "count 6.492999315261841\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      " 83%|████████▎ | 5/6 [01:23<00:15, 15.14s/it]"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "date 10.70549464225769\n",
      "doc 0.0\n",
      "geoname 2.760970115661621\n",
      "count 13.286296606063843\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "100%|██████████| 6/6 [01:45<00:00, 17.49s/it]"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "date 22.202698469161987\n",
      "doc 0.0\n",
      "geoname 7.592491626739502\n",
      "count 18.948084354400635\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "100%|██████████| 6/6 [02:14<00:00, 22.42s/it]"
     ]
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "date 28.867743492126465\n"
     ]
    },
    {
     "output_type": "stream",
     "name": "stderr",
     "text": [
      "\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "source": [
    "print(oaa.iloc[0])"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "admin1_code                       [12]\n",
      "admin2_code                      [035]\n",
      "admin3_code                         []\n",
      "admin4_code                         []\n",
      "location_name               [Américas]\n",
      "location_lat                [18.36455]\n",
      "location_lon               [-99.54467]\n",
      "cases                        [2163354]\n",
      "cases_tags                    [[case]]\n",
      "deaths                            [53]\n",
      "deaths_tags            [[case, death]]\n",
      "dates_start      [2021-01-02 00:00:00]\n",
      "dates_end        [2021-01-03 00:00:00]\n",
      "Name: 0, dtype: object\n"
     ]
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "source": [
    "df['content'].iloc[0]"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "'DENGUE/DHF UPDATE (01): AMERICAS\\n********************************\\nA ProMED-mail post\\n<http://www.promedmail.org>\\nProMED-mail is a program of the\\nInternational Society for Infectious Diseases\\n<http://www.isid.org>\\n\\nIn this update:\\nCases in various countries\\n- WHO/PAHO countries in the Americas reporting dengue cases\\n- WHO epidemiological update\\n\\nMexico and Central America\\n---\\nMexico\\n- National\\n- Hidalgo state\\n- Jalisco state\\nCosta Rica\\n\\nCaribbean\\n---\\nSt Vincent and the Grenadines\\n\\nSouth America\\n---\\nBrazil\\n- National\\n- Rio de Janeiro state\\n- Ribeirao Preto, Sao Paulo state\\n- Minas Gerais state\\n- West central Minas Gerais state\\n- Cascavel, Parana state\\n- Rio Branco, Arce state\\nColombia\\nParaguay\\nPeru\\n\\n******\\nCases in various countries\\n- WHO/PAHO countries in the Americas reporting dengue cases\\nDate: Sat 2 Jan 2021\\n<http://www.paho.org/data/index.php/en/mnu-topics/indicadores-dengue-en/dengue-nacional-en/252-dengue-pais-ano-en.html>\\n\\n\\nReported cases of dengue fever in the Americas, by country or territory to EW 53 of 2020 [week ending (w/e) 2 Jan 2021]\\n-------------------------------------------------------------\\nCases by Country / Week updated / Serotype / Total / Confirmed / Severe / Deaths\\nNorth America\\nUSA / 53 [w/e 2 Jan 2021] / - / 307 / 307 / 0 / 0\\n\\nCentral America and Mexico\\nBelize / 53 [w/e 2 Jan 2021] / D1,3 / 2779 / 324 / 0 / 0\\nCosta Rica 53 [w/e 2 Jan 2021] / D1,2 / 10 056 / 426 / 21 / 0\\nEl Salvador / 52 [w/e 26 Dec 2020] / D2 / 5450 / - / 9 / 0\\nGuatemala / 46 [w/e 14 Nov 2020] / D1,2,3,4 / 5782 / 786 / 14 / 8\\nHonduras / 53 [w/e 2 Jan 2021] / D1,2 / 26 006 / 75 / 1787 / 10\\nMexico / 53 [w/e 2 Jan 2021] / D1,2,3,4 / 120 639 / 24 313 / 1058 / 79\\nNicaragua / 53 [w/e 2 Jan 2021] / D2 / 53 953 / 1259 / 70 / 1\\nPanama / 53 [w/e 2 Jan 2021] / D1 / 4129 / 19162 / 12 / 6\\n\\nLatin Caribbean\\nCuba / 26 [w/e 27 Jun 2020] / D1,2 / 1418 / 1418 / 1 / 0\\nDominican Republic / 53 [w/e 2 Jan 2021] / D1 / 3964 / 453 / 245 / 36\\nPuerto Rico / 53 [w/e 2 Jan 2021] / - / 558 / 558 / 24 / 0\\n\\nNon-Latin Caribbean\\nAnguilla / 32 [w/e 8 Oct 2020] / D1,3 / 6 / 2 / 0 / 0\\nAntigua and Barbuda / 36 [w/ 5 Sep 2020] / D3 / 409 / 3 / 0 / 0\\nAruba / 48 [w/e 28 Nov 2020] / - / 11 / 4 / 0 / 0\\nBahamas / 24 [w/e 13 Jun 2020] / - / 3 / 0 / 0 / 0\\nBarbados / 48 [w/e 28 Nov 2020] / D3 / 301 / 538 / 5 / 0\\nCayman Islands / [53 [w/e 2 Jan 2021] / D2 /35 / 1 / 0 / 0\\nDominica / 53 [w/e 2 Jan 2021] / D2,3 / 88 / 14 / 1 / 0\\nFrench Guiana / 50 [w/e 12 Dec 2020] / D1,2,3 / 8759 / 4287 / 11 / 4\\nGrenada / 53 [w/e 2 Jan 2021] / D3 / 253 / 171 / 4 / 0\\nGuadeloupe / 53 [w/e 2 Jan 2021] / D1,2,3 / 19 407 / 282 / 3 / 2\\nGuyana / 40 [w/e 13 Jun 2020] / - / 403 / 403 / 0 / 0\\nHaiti / 45 [w/e 7 Nov 2020] / - / 17 / 0 / 0 / 0\\nJamaica / 50 [w/e 12 Dec 2020] / - / 824 / 15 / - / 1\\nMartinique / 53 [w/e 2 Jan 2021] / D1,2,3 / 32 398 / 12 276 / 47 / 17\\nSt Barthelemy 53 [w/e 2 Jan 2021] / D1,2 / 1616 / 889 / 0 / 0\\nSt Kitts and Nevis 53 [w/e 2 Jan 2021] / D3 / 20 / 2 / 0 / 0\\nSt Lucia / 53 [w/e 2 Jan 2021] / D2,3 / 13 181 / 1318 / 0 / 0\\nSt Martin / 53 [w/e 2 Jan 2021] / D1,2,3 / 2736 / 488 / 1 / 1\\nSt Vincent and the Grenadines / 53 [w/e 2 Jan 2021] / D3 / 1790 / 1790 / 8 / 8\\nSint Maarten / 53 [w/e 2 Jan 2021] / - / 9 / 9 / 0 / 0\\nSuriname 53 [w/e 2 Jan 2021] / D2,4 / 563 / 277 / 0 / 0\\nTrinidad and Tobago / 53 [w/e 2 Jan 2021] / - / 59 / 2 / 0 / 0\\nTurks and Caicos / 44 [w/e 31 Oct 2020] / D1 / 142 / 23 / 0 / 0\\nVirgin Islands UK / 44 [w/e 31 Oct 2020] / - / 44 / 44 / 0 / 0\\n\\nAndean Subregion\\nBolivia / 53 [w/e 2 Jan 2021] / D1,2, / 85 326 / 14 605 / 223 / 33\\nColombia / 53 [w/e 2 Jan 2021] / D1,2,3,4 / 78 979 / 17 401 / 897 / 55\\nEcuador / 53 [w/e 2 Jan 2021] / D1,2 / 16 570 / - / 51 / 6\\nPeru / 53 [w/e 2 Jan 2021] / D1,2 / 56 394 / 35 641 / 122 / 88\\nVenezuela / 53 [w/e 2 Jan 2021] / D1,2,3 / 6721 / 304 / 31 / 14\\n\\nSouthern Cone\\nArgentina 48 [w/e 28 Nov 2020] / D1,2,4 / 59 358 / 48 415 / 37 / 19\\nBrazil / 46 [w/e 14 Nov 2020] / D1,2,3,4 / 1 418 296 / 750 181 / 799 / 526\\nChile / 53 [w/e 2 Jan 2021] / D2 / 9 / 9 / 0 / 0\\nParaguay / 53 [w/e 2 Jan 2021] / D1,2,4 / 223 782 / 167 / 75\\n\\nTotals for the Americas / 2 251 689 / 992 847 / 5648 / 991\\n\\n[Maps showing the location of the affected islands and countries in the Americas mentioned above and below can be accessed at\\n<http://healthmap.org/promed/p/35574>;\\nNorth America at <http://healthmap.org/promed/p/106>;\\nCentral America at <http://healthmap.org/promed/p/39455>;\\nCaribbean at <http://www.mapsofworld.com/caribbean-islands>; and\\nSouth America at <http://healthmap.org/promed/p/6186.> - Mod.TY]\\n\\n- WHO epidemiological update\\nDate: Thu 3 Dec 2020\\nSource: PAHO/WHO: Dengue in the context of COVID-19 [edited]\\n<https://www.paho.org/en/documents/epidemiological-update-dengue-context-covid-19-3-december-2020>\\n\\n\\nSituation summary\\n-----------------\\nBetween epidemiological week (EW) 1 [w/e 4 Jan 2020] and EW 47 of 2020 [w/e 21 Nov 2020], 2 163 354 dengue cases were reported in the Region of the Americas (incidence of 221.6 cases per 100 000 population), including 872 deaths. Of these cases, 963 787 (45%) were confirmed by laboratory criteria and 5 197 (0.2%) were classified as severe dengue. The case fatality rate was 0.04%, the lowest recorded in the last 10 years, despite being an epidemiologically complex year for dengue due to the simultaneous pandemic transmission of Coronavirus disease (COVID-19).\\n\\nAlthough the cumulative incidence as of EW 47 of 2020 [w/e 21 Nov 2020] for the Region of the Americas is lower than that reported during the epidemic year 2019, it is higher than the incidence reported for the 2016-2018 period and it is the highest reported since 2015 in 2 sub-regions of the Americas: non-Latin Caribbean and Andean Subregion2. (Figure 1)\\n\\nThis situation occurs in parallel with an intense transmission of SARS-CoV-2. As of EW 47 of 2020 [w/e 21 Nov 2020] in the Region of the Americas, the total accumulated cases of COVID-19 were 24 563 640 (2424 cases per 100 000 population) and 697 740 deaths (case fatality rate of 2.8%).\\n\\nIn the current context of the COVID-19 pandemic, the persistence of higher than expected dengue cases in endemic areas and the beginning of the summer season in the southern hemisphere will represent a significant challenge for the concerned populations and the systems that will need to respond to concomitant public health risk events.\\n\\nFurthermore, recent hurricanes and storms have placed an additional burden on the already overburdened health systems in affected Central American countries and have caused the potential slowdown or postponement of vector control activities.\\n\\nFigure 1: Incidence of dengue cases by 100 000 population as of Epidemiological Week (EW) 47 [w/e 21 Nov 2020] by year of report and subregion. Region of the Americas, 2015-2020.\\n\\nAs of EW 47 of 2020 [w/e 21 Nov 2020], 6 of the 10 countries and territories with the highest cumulative incidence of dengue cases correspond to the Non-Latin Caribbean: Saint Barthelemy (15 978 cases per 100 000 population), Saint Martin (7875/100 000), Martinique (7706/100 000), Guadeloupe (3494/100 000), French Guiana (2843/100 000), and Saint Vincent and the Grenadines (1050/100 000).\\n\\nIn the Southern Cone, as of EW 47 of 2020 [w/e 21 Nov 2020], countries with the highest cumulative incidence were Paraguay (3212 cases per 100 000 population), a country that was affected with the largest dengue epidemic in its history, and Brazil (661/100 000), where the number of cases had a decreasing trend since EW 11 of 2020 [w/e 14 Mar 2020]; in the Andean subregion, Bolivia (750/100 000), where cases decreased since EW 9 [w/e 29 Feb 2020] and, in the Central American isthmus, Nicaragua (709/100 000), with a slight upward trend since EW 25 of 2020 [w/e 20 Jun 2020].\\n\\nSituation by subregion\\n----------------------\\n- Non-Latin Caribbean\\nCurrently, the Non-Latin Caribbean subregion is experiencing an active dengue outbreak, with co-circulation of SARS-CoV-2. Dengue cases reported between EW 1 [w/e 4 Jan 2020] and EW 47 of 2020 [w/e 21 Nov 2020] have remained above what was reported for the same period in 2019. In 2020, as of EW 47, 62 620 dengue cases (incidence of 315.4 cases per 100 000 population), including 20 deaths, were reported, with a case fatality rate of 0.03%. In contrast, in 2019, 10 222 dengue cases (51.5 cases per 100 000 population), including 15 deaths, were reported for the same period.\\n\\nThe countries and territories that present dengue cases above the historical trend since 2015 and that have active outbreaks reported in the 3-week period of EW 43 [w/e 24 Oct 2020] to EW 45 of 2020 [w/e 7 Nov 2020] are Martinique (with an average of 1058 cases per week), Guadeloupe (with an average of 990 cases per week), French Guiana (with an average of 87 cases per week), and Saint Barthelemy (with an average of 32 cases per week).\\n\\nWith regard to the COVID-19 pandemic, a 1st wave among these countries and territories started in EW 20 of 2020 [w/e 16 May 2020] and a 2nd one started in EW 31 [w/e 1 Aug 2020], coinciding with an increase in notifications of dengue cases (Figure 2). Of the 6 countries and territories with the highest dengue incidences, French Guiana, Guadeloupe, Martinique, and Saint Martin also report community transmission of COVID-19, while Saint Barthelemy and Saint Vincent and the Grenadines report sporadic cases of COVID-19.\\n\\nDue to this situation, there has been substantial technical cooperation over the last 3 months in this subregion, prioritizing patient management as a method to prevent severe cases and deaths, and the case fatality rate of 0.03% has been maintained.\\n\\nFigure 2: Distribution of dengue and COVID-19 cases by epidemiological week (EW). Non-Latin Caribbean, EW 1 to 47 of 2019 and 2020.\\n\\n- Andean Subregion\\nBetween EW 1 [w/e 4 Jan 2020] and EW 47 of 2020 [w/e 21 Nov 2020], 226 970 cases of dengue (incidence of 159.3 cases per 100 000 population), including 153 deaths were reported in the Andean Subregion. The case fatality rate was 0.07%.\\n\\nThis subregion reported the highest number of reported dengue cases between EW 1 [w/e 4 Jan 2020] and EW 17 of 2020 [w/e 25 Apr 2020]. The number of reported cases peaked in EW 7 of 2020 [w/e 15 Feb 2020] and began to decrease steadily from EW 10 [w/e 7 Mar 2020], coinciding with the increase of COVID-19 cases. From EW 20 [w/e 16 May 2020] to EW 47 [w/e 21 Nov 2020], the number of cases registered per week remained below those reported in 2019. In this subregion, the number of COVID-19 cases reached a peak in EW 34 of 2020 [w/e 22 Aug 2020]. (Figure 3)\\n\\nBolivia reported an unusual increase in dengue cases since EW 2 of 2020 [w/e 11 Jan 2020] that peaked in EW 7 [w/e 15 Feb 2020], with 8976 cases reported weekly. After EW 8 [w/e 22 Feb 2020], cases began to decline, and as of EW 25 [w/e 20 Jun 2020] there continued to be less than 100 cases reported per week.\\n\\nIn the 3-week period of EW 45 [w/e 7 Nov 2020] to EW 47 [w/e 21 Nov 2020], Colombia maintained an average of 623 reported cases per week, the lowest average since EW 18 of 2018 [w/e 5 May 2018] and after 2 years with high-intensity circulation, having reached a peak in EW 3 of 2020 [w/e 18 Jan 2020] with 4947 weekly cases reported.\\n\\nFigure 3: Distribution of dengue and COVID-19 cases by epidemiological week (EW). Andean Subregion, EW 1 to 47 of 2019 and 2020.\\n\\n- Southern Cone\\nBetween EW 1 [w/e 4 Jan 2020] and EW 47 of 2020 [w/e 21 Nov 2020], 1 661 700 cases of dengue (incidence of 589.8 cases per 100 000 population), including 591 deaths, were reported in the Southern Cone. The case fatality rate was 0.04%.\\n\\nThis subregion reported a sustained increase in the number of dengue cases between EW 1 [w/e 4 Jan 2020] and EW 11 of 2020 [w/e 14 Mar 2020], followed by the reported weekly average falling to almost half since EW 12 [w/e 21 Mar 2020]; this coincides with the start of the COVID-19 pandemic and measures restricting mobility applied by countries to contain the pandemic. Since EW 17 of 2020 [w/e 25 Apr 2020], there has been a sustained decrease in the number of notifications of dengue cases.\\n\\nIn this subregion, the number of accumulated dengue cases and the cumulative incidence as of EW 47 [w/e 21 Nov 2020], did not reach the epidemic figures reported in 2019, even with the outbreak reported in the 1st quarter of 2020. In 2019, there were 2 203 396 cases reported during the same period, with an incidence of 782 cases per 100 000 population. (Figure 4)\\n\\nNotably, Paraguay reported a historical peak of suspected dengue cases with 33 468 cases in EW 6 of 2020 [w/e 8 Feb 2020]; in addition, in the same week, the highest peak of confirmed dengue cases was reported with 9350 cases. Since EW 12 [w/e/ 21 Mar 2020], the number of dengue cases per week has dropped to less than 1000 cases weekly. In the 3-week period of EW 45 [w/e 7 Nov 2020] to EW 47 [w/e 21 Nov 2020], the weekly average of notifications of suspected cases was 183. As of EW 47 of 2020, a total of 61 573 cases (confirmed and probable) were reported with an incidence rate of 849 cases per 100 000 population.\\n\\nFigure 4: Distribution of dengue and COVID-19 cases by epidemiological week (EW). Southern Cone, EW 1 to 47 of 2019 and 2020.\\n\\n- Latin Caribbean\\nBetween EW 1 [w/e 4 Jan 2020] and EW 47 of 2020 [w/e 21 Nov 2020], 5592 dengue cases (incidence of 21.5 cases per 100 000 population), including 35 deaths, were reported in the Latin Caribbean subregion. The case fatality rate (0.63%) is the highest in the region of the Americas, with the Dominican Republic being the country that reports all of the deaths for this subregion. With respect to COVID-19, since EW 12 of 2020 [w/e 21 Mar 2020], there has been an increase in the number of reported cases, with a peak in EW 31 of 2020 [w/e 1 Aug 2020]. (Figure 5)\\n\\nIn this subregion, as of EW 47 of 2019 [w/e 21 Nov 2019], 20 896 dengue cases (incidence of 80.3 cases per 100 000 population), including 47 deaths, were reported, with a case fatality rate of 0.23%.\\n\\nFigure 5: Distribution of dengue and COVID-19 cases by epidemiological week (EW). Latin Caribbean, EW 1 to 47 of 2019 and 2020.\\n\\n- Central American Isthmus and Mexico\\nBetween EW 45 [w/e 7 Nov 2020] and EW 47 of 2020 [w/e 21 Nov 2020], 2 successive hurricanes affected the Central American Isthmus, mainly Guatemala, Honduras, and Nicaragua, causing floods, landslides, and damage to infrastructure, homes, and services. The consequent high demand in health and social services increases the risk of disease transmission, as well as makes epidemiological surveillance and vector control difficult.\\n\\nBetween EW 1 [w/e 4 Jan 2020] and EW 47 of 2020 [w/e 21 Nov 2020], 206 211 dengue cases (incidence of 115 cases per 100 000 population), including 73 deaths, were reported in the Central American Isthmus and Mexico. The case fatality rate was 0.04%.\\n\\nThe number of reported dengue cases has remained stable as of EW 47 of 2020 [w/e 21 Nov 2020] and below that reported for the same period in 2019, when 623 980 cases (incidence of 347.9 cases per 100 000 population), including 415 deaths, were reported. Dengue cases are being reported simultaneously to the appearance and increase of COVID-19 cases, which had peaks in EW 30 [w/e 25 Jul 2020] and EW 41 [w/e 10 Oct 2020], followed by another peak in COVID-19 cases in EW 48 of 2020 [w/e 28 Nov 2020]. (Figure 6)\\n\\nAs of EW 45 of 2020 [w/e 7 Nov 2020], Costa Rica had reported 9556 cases of dengue and zero deaths. An increase in dengue cases had been reported between EW 19 [w/e 9 May 2020] and EW 32 [w/e 8 Aug 2020], reaching a peak of 451 weekly cases in EW 27 [w/e 4 Jul 2020], and then decreasing in the following weeks.\\n\\nIn the 3-week period of EW 45 [w/e 7 Nov 2020] to EW 47 [w/e 21 Nov 2020], 2 countries had reported sustained circulation: Mexico and Nicaragua. Mexico reported an average of 3858 cases per week, similar to the trend reported between 2016 and 2018, but lower than the 2019 epidemic year, when an average of 10 040 cases were reported per week. Nicaragua reported an average of 1130 cases per week, below its epidemic threshold.\\n\\nFigure 6: Distribution of dengue and COVID-19 cases by epidemiological week (EW). Central American Isthmus and Mexico, EW 1 to 48 of 2019 and 2020.\\n\\nAdvice to member states\\n-----------------------\\nGiven the coexistence of COVID-19 with dengue and other arboviruses in several countries and territories of the Region of the Americas, the Pan American Health Organization/World Health Organization (PAHO/WHO) urges member states to continue strengthening surveillance, diagnosis, triage, and adequate treatment of dengue in the context of COVID- 19, while establishing strategies to facilitate access to health services for patients with dengue and other arboviruses. PAHO/WHO recommends the adequate triage of patients both for the timely detection of dengue warning signs and to reduce the risk of SARS-CoV-2 infections acquired in health care services.\\n\\nPAHO/WHO reminds member states of the guidelines published in the 10 Jun 2020 PAHO/WHO epidemiological update on dengue and other arboviruses, available at <https://bit.ly/3mM1H36>; as well as those related to COVID-19 published in the 9 Nov 2020 PAHO/WHO Epidemiological Update, available at <https://bit.ly/3g4vK3S>.\\n\\n--\\ncommunicated by:\\nProMED rapporteur Marianne Hopp\\n\\n[Maps showing the location of the affected islands and countries in the Americas mentioned above and below can be accessed at\\n<http://healthmap.org/promed/p/35574>;\\nNorth America at <http://healthmap.org/promed/p/106>;\\nCentral America at <http://healthmap.org/promed/p/39455>;\\nCaribbean at <http://www.mapsofworld.com/caribbean-islands>; and\\nSouth America at <http://healthmap.org/promed/p/6186>. - Mod.TY]\\n\\nCases in various countries not mentioned above or more recent case numbers:\\nMexico and Central America\\n---\\nMexico\\n- National. 22 Jan 2021. (reported) for 2020, 120 239 cases.\\n<http://outbreaknewstoday.com/brazil-paraguay-mexico-vietnam-and-malaysia-report-the-most-dengue-cases-in-2020/>\\n\\n- Hidalgo state. 19 Dec 2020. (registered) 764 cases; DHF/serious 4 cases.\\n<https://www.milenio.com/ciencia-y-salud/dengue-hidalgo-entidad-registra-764-casos> [in Spanish]\\n\\n- Jalisco state. 10 Jan 2021. (probable) 23 166 cases, (confirmed) 5362 cases; deaths (confirmed) 20, under investigation 18 cases. Municipalities most affected: Tlaquepaque 653 cases, Guadalajara 483, Zapopan 367, Jocotepec 313, and Tlajomulco 294.\\n<https://www.vallartadaily.com/jalisco-leads-the-country-in-dengue-cases/>\\n\\nCosta Rica. 21Nov 2020. (registered) 9438 cases. Localities most affected: Huetar Caribe 4353 cases and La Brunca 2228 cases.\\n<https://qcostarica.com/dengue-cases-in-2020-already-exceed-all-cases-in-2019/>\\n\\nCaribbean\\n---\\nSt Vincent and the Grenadines. 24 Nov 2020. (confirmed) 1617 cases; deaths 7.\\n<http://jamaica-gleaner.com/article/caribbean/20201124/child-dies-st-vincent-dengue-fever>\\n\\nSouth America\\n---\\nBrazil\\n- National. 22 Jan 2021. (reported) for 2020, 1 418 296 cases.\\n<http://outbreaknewstoday.com/brazil-paraguay-mexico-vietnam-and-malaysia-report-the-most-dengue-cases-in-2020/>\\n\\n- Rio de Janeiro state. 27 Nov 2020. (reported) 31 210 cases, significant drop in cases.\\n<http://outbreaknewstoday.com/brazil-dengue-cases-in-rio-de-janeiro-state-decrease-significantly-in-2020/>\\n\\n- Ribeirao Preto, Sao Paulo state. 22 Nov 2020. (registered) 17 538 cases to October 2020, more than the 14 421 cases last year (2019).\\n<http://outbreaknewstoday.com/brazil-ribeirao-preto-dengue-cases-top-17k-exceeds-all-of-2019/>\\n\\n- Minas Gerais state. 23 Dec 2020. (suspected) 26 526 cases (confirmed) 57 861 cases; deaths (confirmed) 12 (under investigation) 57.\\n<https://andradas.portaldacidade.com/noticias/saude/andradas-tem-mais-tres-casos-suspeitos-de-dengue-registrados-pela-ses-0628> [in Portuguese]\\n\\n- West central Minas Gerais state. 31 Dec 2020. (confirmed) 10 390 cases. Municipalities most affected: Minas Gerais 3636 cases, Bom Despacho 784 cases.\\n<https://g1.globo.com/mg/centro-oeste/noticia/2020/12/31/mais-de-10-mil-casos-provaveis-de-dengue-foram-registrados-em-2020-no-centro-oeste-de-mg-segundo-boletim-do-estado.ghtml> [in Portuguese]\\n\\n- Cascavel, Parana state. 29 Dec 2020. (reported) 72 cases.\\n<https://cgn.inf.br/noticia/320696/casos-de-dengue-aumentam-em-cascavel> [in Portuguese]\\n\\n- Rio Branco, Arce state. 2 Feb 2021. (reported) 1492 cases. Health emergency decreed; DHF/serious 6 cases, deaths 2.\\n<http://outbreaknewstoday.com/brazil-mayor-of-rio-branco-issues-emergency-declaration-over-dengue-fever-26685/>\\n\\nColombia. 21 Dec 2020. (reported) as of week 50 [w/e 12 Dec 2020], deaths (probable) 179. Based historically, departments with the expected number of cases: Amazonas, Atlantico, Barranquilla, Arauca, Bolivar, Casanare, Caqueta, Caldas, Cordoba, Guainia, Cesar, Huila, La Guajira, Guaviare, San Andres, Providencia, and Santa Catalina, Quindio, Risaralda, Norte de Santander, Santander, Meta, Magdalena, Sucre, Narino, and Tolima. Departments with a greater number of cases than expected: Cauca, Choco, Cartagena, and Putumayo; Santa Marta municipality 388 cases with 35 DHF/serious cases and 3 deaths; Magdalena department 603 cases with 16 DHF/serious cases and 2 deaths.\\n<https://www.opinioncaribe.com/2020/12/21/alerta-por-388-casos-de-dengue-en-santa-marta/> [in Spanish]\\n\\nParaguay. 22 Jan 2021. (reported) for 2020, 223 082 cases.\\n<http://outbreaknewstoday.com/brazil-paraguay-mexico-vietnam-and-malaysia-report-the-most-dengue-cases-in-2020/>\\n\\nPeru. 15 Jan 2021. (reported) 56 394 cases in 2020 the worst outbreak since 2017; deaths 88.\\n<http://outbreaknewstoday.com/peru-reports-worst-dengue-epidemic-since-2017/>\\n\\n--\\ncommunicated by:\\nProMED from HealthMap Alerts\\n<promed@promedmail.org>\\n\\n[See Also:\\n2020\\n---\\nDengue/DHF update (12): Americas 20201115.7945357\\nDengue/DHF update (10): Americas 20200901.7737065\\nDengue/DHF update (09): Americas 20200811.7664969\\nDengue/DHF update (07): Americas 20200618.7482266\\nDengue/DHF update (05): Americas 20200429.7278906\\nDengue/DHF update (04): Americas 20200330.7159571\\nDengue/DHF update (03): Americas 20200307.7058361\\nDengue/DHF update (01): Americas 20200114.6889091]\\n.................................................sb/lm/ml/jh/mpp/sh/ty/mj/sh'"
      ]
     },
     "metadata": {},
     "execution_count": 92
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [
    "['admin1_code',\r\n",
    "'admin2_code',\r\n",
    "'admin3_code',\r\n",
    "'admin4_code',\r\n",
    "'location_name',\r\n",
    "'location_lat',\r\n",
    "'location_lon',\r\n",
    "'cases',\r\n",
    "'cases_tags',\r\n",
    "'deaths',\r\n",
    "'deaths_tags',\r\n",
    "'dates_start',\r\n",
    "'dates_end',]"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "source": [
    "df= pd.read_feather('combined_df_anomaly.feather')"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "source": [
    "bruh = df[~df['content'].str.contains('|'.join(('case', 'cases', 'death', 'deaths')))].reset_index()"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "source": [
    "df = df[df['content'].str.contains('|'.join(('case', 'cases', 'death', 'deaths')))].reset_index()"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [],
   "outputs": [],
   "metadata": {}
  }
 ],
 "metadata": {
  "orig_nbformat": 4,
  "language_info": {
   "name": "python",
   "version": "3.9.6",
   "mimetype": "text/x-python",
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "pygments_lexer": "ipython3",
   "nbconvert_exporter": "python",
   "file_extension": ".py"
  },
  "kernelspec": {
   "name": "python3",
   "display_name": "Python 3.9.6 64-bit ('geopandas': conda)"
  },
  "interpreter": {
   "hash": "3387528f949e08a7465409a492fe128ebdb03217a22eb98a1ae9d56d862b3753"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}